### PR TITLE
Added pavement to the Tiberian Sun mod

### PR DESCRIPTION
**Pull request by Mailaender**
It isn't rendered for reasons unknown to me.

### CODE REVIEW

File: mods/ts/rules/structures.yaml [hunk 1]
```diff
@@ -169,6 +169,29 @@ GAPILE:
...
+		Adjacent: 4
+		TerrainTypes: Clear, DirtRoad, Rough
+		BuildSounds: place2.aud
+		AllowInvalidPlacement: true
+		Footprint: xx xx
+		Dimensions: 2,2
+	KillsSelf:
+		RemoveInstead: true
+	LaysTerrain:
+		Template: 671
```
Comment (pchote) [inline mods/ts/rules/structures.yaml]: TS LATs the concrete (smooths the edges, like we do with D2K spice) using 9 different templates, which means this simple approach isn't really going to work.  The simplest way I can see is to have a specific TSConcrete layer that accepts a bool (is concrete yes/no), and then picks the right template to use for each tile based on its neighbours.

Comment (penev92) [inline mods/ts/rules/structures.yaml]: I don't see how this is different from what we do for D2k and hence how it justifies adding new completely unrelated code that does the same thing.

Comment (pchote) [inline mods/ts/rules/structures.yaml]: It is different from D2K because D2K doesn't do LAT-style concrete.

Comment (Mailaender) [inline mods/ts/rules/structures.yaml]: The problem is that I can't even do non-LAT-style pavement in TS.

Comment (penev92) [inline mods/ts/rules/structures.yaml]: Also D2k has different concrete templates as well. Us not using them is more of an oversight/attempt not to clutter the map editor's tile picker.

Comment (pchote) [inline mods/ts/rules/structures.yaml]: That is quite different.  Compare D2KResourceLayer (which does LAT tile selection for spice - like TS concrete) to the regular ResourceLayer (which uses multiple variants of a single full-cell tile - like D2K concrete).

**Comment by tomsons26** [comment]
idkwid lol but
LaysTerrain:
	Template: 671
	TerrainTypes: Clear, DirtRoad, Rough, Road
makes it actually show that one tile 671 has

**Comment by Mailaender** [comment]
Okay, now I get it. 

```
	LaysTerrain:
		TerrainTypes:
```

is again (redundant to Building) the TerrainType where the game is allowed to place it, not the terrain type to place.

Works now in placeholder quality for TEMPERAT as the terrain IDs differ to SNOW for whatever reason.

**Comment by Phrohdoh** [comment]
Perhaps that should be renamed, for consistency, to `ValidTerrainTypes`.

**Comment by obrakmann** [comment]
Works in the temperate map, but in the snow one, while the preview is 2x2 cells, the placed block is 1x1.

**Comment by LipkeGu** [comment]
the pavement is 5x5 maybe we could enable line build (via holding shift) as an extra feature :)

**Comment by pchote** [comment]
I outlined how to fix this in one of the now-outdated diff trees:

> The simplest way I can see is to have a specific TSConcrete layer that accepts a bool (is concrete yes/no), and then picks the right template to use for each tile based on its neighbours.

**Comment by SMIFFGIG** [comment]
I would strongly suggest that pavement is built underneath structures when placed too.
This was such an annoyance in TS as you could pave your entire base but when a structure exploded it would still cause a huge crater (making it impossible to build there due to deformed terrain).

It also looked ugly.
